charts/newrelic-infrastructure-beta/templates/kubelet/daemonset.yaml

```diff
@@ -0,0 +1,273 @@
+            {{- if (include "newrelic.customAttributes" .) }}
+            - name: "NRIA_CUSTOM_ATTRIBUTES"
+              value: {{ include "newrelic.customAttributes" . }}
+            {{- end }}
+            {{- if .Values.verboseLog }}
+            - name: NRIA_VERBOSE
+              value: "1"
+            {{- end }}
+            {{- with .Values.logFile }}
+            - name: NRIA_LOG_FILE
+              value: {{ . }}
+            {{- end }}
+            {{- if not .Values.privileged }}
+            - name: "NRIA_OVERRIDE_HOST_ROOT"
+              value: ""
+            - name: "NRIA_IS_SECURE_FORWARD_ONLY"
+              value: "true"
+            {{- end }}
+            {{- with .Values.kubelet.extraEnv }}
+            {{- toYaml . | nindent 12 }}
+            {{- end }}
+          {{- with .Values.kubelet.extraEnvFrom }}
+          envFrom: {{ toYaml . | nindent 12 }}
+          {{- end }}
+          volumeMounts:
+            {{- if .Values.config }}
+            - name: config
+              mountPath: /etc/newrelic-infra.yml
+              subPath: newrelic-infra.yml
+            {{- end }}
+            {{- if .Values.integrations_config }}
+            - name: nri-integrations-cfg-volume
+              mountPath: /etc/newrelic-infra/integrations.d/
+            {{- end }}
+            {{- if .Values.privileged }}
+            - name: dev
+              mountPath: /dev
+            - name: host-docker-socket
+              mountPath: /var/run/docker.sock
+            - name: log
+              mountPath: /var/log
+            - name: host-volume
+              mountPath: /host
+              readOnly: true
+            {{- else }}
+            - mountPath: /var/db/newrelic-infra/data
+              name: tmpfs-data
+            - mountPath: /var/db/newrelic-infra/user_data
+              name: tmpfs-user-data
+            - mountPath: /tmp
+              name: tmpfs-tmp
+            {{- end }}
+            {{- with .Values.kubelet.extraVolumeMounts }}
+            {{- toYaml . | nindent 12 }}
+            {{- end }}
+          {{- with .Values.kubelet.resources }}
+          resources: {{ toYaml . | nindent 12 }}
+          {{- end }}
+      volumes:
+        {{- if .Values.privileged }}
+        - name: dev
+          hostPath:
+            path: /dev
+        - name: host-docker-socket
+          hostPath:
+            path: /var/run/docker.sock
+        - name: log
+          hostPath:
+            path: /var/log
+        - name: host-volume
+          hostPath:
+            path: /
+        {{- else }}
+        - name: tmpfs-data
+          emptyDir: {}
+        - name: tmpfs-user-data
+          emptyDir: {}
+        {{- end }}
+        - name: tmpfs-tmp
+          emptyDir: {}
+        - name: nri-kubernetes-config
+          configMap:
+            name: {{ template "newrelic.fullname" . }}-kubelet
+            items:
+              - key: nri-kubernetes-config.yml
+                path: nri-kubernetes-config.yml
```

Comment:
Could we avoid using a simple file from the `configMap`? If we could mount the whole `configMap` as a filder with its files, the cluster will eventually change the file if it is changed by hand.

If not we should use an annotation with the config related to each integration hashed to force the restart of the pod: https://helm.sh/docs/howto/charts_tips_and_tricks/#automatically-roll-deployments